### ISSUE TITLE
EX - Split Private Development or CIP sub/work types into separate groups (SS) (UAT in PROD)

### ISSUE THREAD

**Issue by atdservicebot** [cityofaustin/atd-data-tech#14818]
**Acceptance Criteria**
- For the EX folder, Subtype 'Private Development or CIP' needs to be removed as an option for any new applications moving forward.
- Instead, we need two new subtypes: 'Private Development' and 'Capital Improvement Project (CIP)'. 
- The fee workflow will be different for each subtype:
    - Private: system should prompt the portal applicant to pay the permit fee, and the folder status will update to Pending Payment
    - CIP: System will insert fees but not bill and update the folder status to Internet Pending. The PLA will manually invoice the fees, and the folder status will update to Internet Pending.
- For the CIP subtype, Make 'CIP ID', 'CIP Project Manager', 'CIP excavation inspector' info fields mandatory (required to insert) and remove 'CIP?' info field. 

### Workflow

1. Customer submits application
2. ❗️System inserts correct application fee & tech surcharge `Application fee new or emergency application fee`
3. ❗️System updates folder status to Pending Payment
4. ❗️Customer pays the bill
5. System updates folder status to Internet Pending
6. System updates folder in-date to sysdate

**Notes:**
1. Everything other than fees will be default EX behavior (such as infos, attachments, people rules, issuance rules...)

![image](https://github.com/user-attachments/assets/a0fbe17b-1126-48b8-8c0e-6d84dc0a5282)
----

> What application are you using?

AMANDA

> Describe the problem.

For EX and RW-TURP folders, ROW wants to break down the "Private Development or CIP" Sub Type/Work Type into separate Sub Types/Work Types preliminarily titled "Private Development" and "Capital Improvement Project (CIP)". Both Sub Types/Work Types will require changes.

> Describe the outcome you'd like to see when this feature is implemented.

These two applicant types operate differently and deserve to have unique Sub Types/Work Types to account for the differences and minimize confusion.

> Describe any workarounds you currently have in place or alternative solutions you've considered.

N/A

> Requested By
Kyle K.

Request ID: DTS23-111260

**Comment by KyleErich** [comment]
@TracyLinder Bumping this card based on the discussion we had today in the 6/14 DTS-ROW meeting. DSD Inspections identified this as a problem for them too, as it causes confusion between them and the CIP inspections handled by the Construction Management Division. Hoping we can get this on deck, I think I'll change the need to a Should Have.

**Comment by TracyLinder** [comment]
**We need to:**
- Document the workflow for each
- Update procedures with the correct sub/work codes

**Comment by vschandramouli** [comment]
Notes from scoping 10/17:

Needing this split is mainly for tracking and semantics, workflow may not be too different. 

Make CIP ID, CIP Project Manager, CIP excavation inspector info field mandatory in CIP subtype and remove 'CIP?' info field.

**Comment by vschandramouli** [comment]
Updated AC, moving to Backlog.

**Comment by SubhaShroff** [comment]
@Nadin-Nader @vschandramouli 
We already have both subtypes and using for UC. We can use the same for EX also. But, with Private Development there is '/Utility' added. Want to ask if we can use the same 'Private Development/Utility' or want to create one with just 'Private Development'.
50689	Private Development/Utility
51600	Capital Improvement Project (CIP)

Thanks.

**Comment by TracyLinder** [comment]
**Conversation with ROW:**
- Tracy thinks EX should have it's own subtypes & not share with UC
- Bond doesn't want the word 'utility' in the EX subtype description
- Paloma would like Izzy to weigh in on this; she isn't in attendance.

**Comment by SubhaShroff** [comment]
@Nadin-Nader SCTASK0469375 is created a DEV request with Priya's help. Thank you.

**Comment by SubhaShroff** [comment]
@Nadin-Nader  @KyleErich @vschandramouli 
'CIP excavation inspector' - I think its  Excavation Inspector (76155). Please confirm. Thanks.

**Comment by KyleErich** [comment]
@SubhaShroff @Nadin-Nader @vschandramouli @TracyLinder I understand. Let's not make any name changes. Is it possible (on a separate card) to create a new info field and have it apply only to this one sub type (EX-CIP)? If so, I'd like to submit a separate card to add a new CIP Inspector info field with its own dropdown options and remove Excavation Inspector 76155 from just this one EX sub type.

**Comment by SubhaShroff** [comment]
@KyleErich Since subtype is also new, I think we can create a new info. You can add that too in the same card AC. So, it would help to add the new info code into the validation for it, which is already mentioned in this ticket AC.  
CC: @Nadin-Nader @vschandramouli @TracyLinder

**Comment by SubhaShroff** [comment]
@Nadin-Nader  SCTASK0474111  is submitted for DEV deployment of the changes that were suggested in code review. Thank you.

**Comment by SubhaShroff** [comment]
@Nadin-Nader I tested it on DEV. Please create a deployment request. Thank you.
[14818 -  TEST  Deployment Plan.docx](https://github.com/user-attachments/files/18028537/14818.-.TEST.Deployment.Plan.docx)

**Comment by Nadin-Nader** [comment]
This has been sent to The DevOps CM Team to be deployed to TEST under **SCTASK0474462**

**Comment by Nadin-Nader** [comment]
This will be worked on by Monty at CTM in a meeting with Subha and myself. Waiting on Monty to set a time. CTM DevOps Team TASK number is **SCTASK0476364**

**Comment by SubhaShroff** [comment]
@Nadin-Nader  I verified the code on TEST and tested. Here are the test results. Its ready for UAT. Thank you.
[TEST env - 14818 test cases.xlsx](https://github.com/user-attachments/files/18171371/TEST.env.-.14818.test.cases.xlsx)

### NOTE: 

1. Not verified for other processes.
2. Following documents were added with permits submission for both subtypes. I am not sure if that is anticipated.
![image](https://github.com/user-attachments/assets/d6640753-abf8-44da-86bb-2060abf8a530)

**Comment by KyleErich** [comment]
@SubhaShroff @Nadin-Nader @TracyLinder @vschandramouli Let's pump the brakes a bit on this one. I started UAT on Test and noticed some problems when I tried applying through each of the new pathways on AB+C Test. Going to put this on the agenda for Friday to discuss further about what needs to go in this ticket and what needs to be submitted separately.

No acknowledgements present:
![Private Dev 1](https://github.com/user-attachments/assets/1473cc02-d7f9-462b-a46d-202fd3e8166e)

Correct acknowledgements:
![Old Private Dev 1](https://github.com/user-attachments/assets/b570247d-377f-4464-9c3a-f789981223ef)

No explanatory text for Project or Event Name & Description:
![Private Dev 2](https://github.com/user-attachments/assets/35c07860-cd07-4862-91cc-5325b064b240)

Correct explanatory text:
![Old Private Dev 2](https://github.com/user-attachments/assets/bf6749ba-1896-4408-9966-15c6198bc345)

Info fields are showing a bunch of old fields that aren't on the current AB+C Test portal:
![Cap Imp 1](https://github.com/user-attachments/assets/f728386c-dc2d-4c3b-bb91-132cacc0278f)
![Cap Imp 2](https://github.com/user-attachments/assets/5f72134a-0331-4609-891d-1e6631d2dbc7)
![Cap Imp 3](https://github.com/user-attachments/assets/d0621a39-19d9-409b-ab68-6e2b62f82fd3)

**Comment by TracyLinder** [comment]
### Conversation with ROW at Friday Mtg 12/20/24

- We need to preserve the ability for internal users to query the historical "Private Development or CIP" subtype.
- Kyle's feedback above is understood and Priya will work with Swapna

**Comment by SubhaShroff** [comment]
@Nadin-Nader  @TracyLinder I listed the following places it needs fixing based on Kyle's above feedback. I have been trying to get Swapna's time. She has asked me to schedule a call next week as she is working on some important stuff. Thank you.

**1. Acknowledgement - Verified the possible lookup table looks good. But not displaying acknowledgement. Will need Portal Dev team's help.**
_(For Dev notes: SELECT * FROM validlookup WHERE lookupcode = 50240 AND lookup2 = 50540; -- Acknowledgement - webservice - 83907; - Returns two records.)_
**2. TCP field missing - Verified the possible lookup table looks good. But not displaying acknowledgement. Will need Portal Dev team's help.**
_(For DEV notes: SELECT * FROM validlookup WHERE lookupcode = 83942; -- TCP display webservice - 83972; )_
**3. Freeform - unwanted fields are displaying. I will fix this.**
_( For DEV notes: select * from validlookup vl3 where vl3.lookupcode = 50226 AND LOOKUP2 = 50540; -- Webservice - 83941 - list of unwanted freeform fields)_

**Comment by TracyLinder** [comment]
- Subha is meeting with Swapna on Monday 1/6/25

**Comment by TracyLinder** [comment]
![image](https://github.com/user-attachments/assets/54dbfcca-1962-4c43-9355-e16b4a350216)

**Comment by TracyLinder** [comment]
@MirUlfath reminder to send Subha documentation of the workflow.

**Comment by TracyLinder** [comment]
- Subha and Swapna met yesterday; Swapna was unable to resolve the acknowledgements issue so she will call the vendor today for a solution.

**Comment by SubhaShroff** [comment]
@Nadin-Nader  @KyleErich  @vschandramouli  We have the following help text for Private Development or CIP. Are we going to have the same text for both split subtypes?
![image](https://github.com/user-attachments/assets/cc904f1e-b3e2-4ac0-a389-f2887342af3f)

**Comment by Nadin-Nader** [comment]
This has been sent to the DevOps CM Team to be deployed to DEV under **SCTASK0481022**

**Comment by Nadin-Nader** [comment]
@SubhaShroff this has been deployed ot DEV please UAT

**Comment by SubhaShroff** [comment]
@Nadin-Nader Testing the DEV deployment:
PORTAL --
**1. Acknowledgement:**  Displaying.
**2. TCP field missing** - Displays
**3. Freeform** - unwanted fields removed.
**4. Required People Types:** -
**5. Online Payment:** -  Online payment can only be tested in TEST.

AMANDA --
**1. Preserve the ability for internal users to query the historical "Private Development or CIP" subtype**:
**2. Fees:**
**3. Payment AMANDA :**
**4. Issue:**
**5.  Statuses:**

Testing for full cycle as mentioned in AC... ... **IN PROGRESS..**

As mentioned in the Friday 2pm call, need to add the separate subtype description for each subtype. Worked on it need to get it deployed in DEV.

**Comment by SubhaShroff** [comment]
@Nadin-Nader 
As mentioned above, a commit is done to add subtype description. And also, to fix the default proc of EX folder initialization.

SCTASK0482489 - is created for DEV deployment. Thank you.

**Comment by Nadin-Nader** [comment]
Asked CTM (Hari) if he has deployed this. waiting on response

**Comment by SubhaShroff** [comment]
@Nadin-Nader DEV deployment verified. 
### 1. Subtype description is fixed.
![image](https://github.com/user-attachments/assets/67a0377b-ed5f-4be7-98c3-a4d976941c98)
![image](https://github.com/user-attachments/assets/1e4c7fc9-bcc3-4d06-b560-9148b60b0ac7)

### 2. Folder Initialize procedure is still not reflecting, though the deployment looks good. 
As of now I could think that the default procedure on DEV needs a manual compilation to resolve this. We might need Jence/CTM help for this.
Checking if any other possible reasons for this issue before reaching out to them.
Thanks.

**Comment by Nadin-Nader** [comment]
This other deployment has been sent to the DevOps CM Team to be deployed to DEV under **SCTASK0484007**

**Comment by SubhaShroff** [comment]
@Nadin-Nader  After fixing defaultprocess by Yuan, it all looked good on DEV. SCTASK0484988 - is requested for TEST deployment. Thanks.

**Comment by SubhaShroff** [comment]
@Nadin-Nader 
### TEST testing is in PROGRESS... 

![Image](https://github.com/user-attachments/assets/d261e068-1112-4380-a987-3c6737746504)

**Comment by SubhaShroff** [comment]
@Nadin-Nader This is deployed in TEST. I have some test cases created and tested. And I request a UAT on TEST on call, so I can be part of it and we make sure we have covered every scenario.
Thank you.

[TEST env - 14818 test cases - Results2.xlsx](https://github.com/user-attachments/files/18649373/TEST.env.-.14818.test.cases.-.Results2.xlsx)

**Comment by vschandramouli** [comment]
@SubhaShroff @Nadin-Nader ROW team wants to tackle this as part of the Friday meeting. I will add this to the meeting agenda for 2/7

**Comment by vschandramouli** [comment]
Feedback from ROW Team based on UAT done on 2/7/25

- Remove the following fields from Private EXs:
CIP - Yes/No?
CIP ID #
CIP Project Manager
 
- Make Excavation Inspector info field non-Mandatory in CIP subtype

**Comment by SubhaShroff** [comment]
@Nadin-Nader @TracyLinder  I gave a demo of this ticket in 2pm, Friday call. There few points to do changes. @vschandramouli  please add the points here.
@KyleErich @BondRuss Here is the testcases with results on it. Please review 'TEST-2' tab/workbook on excel sheet. Please UAT on TEST.

[TEST env - 14818 test cases - Results2.xlsx](https://github.com/user-attachments/files/18712777/TEST.env.-.14818.test.cases.-.Results2.xlsx)

Thank you.
----------------------------------------
@Nadin-Nader  SCTASK0489082 is submitted for DEV deployment of 

- removing CIP fields for Private Deployment
- Excavation Manager is made non-mandatory for CIP.

**Comment by BondRuss** [comment]
@SubhaShroff @KyleErich @TracyLinder I think we want to go through the UAT live on Friday during the DTS meeting again since there are so many stakeholders affected within ROW. Hopefully, this is agreeable. Thank you for your work on this.

**Comment by SubhaShroff** [comment]
@Nadin-Nader  I tested the DEV deployment of SCTASK0489082. Please create a TEST deployment request for this. Thank you.

[14818 - b8574c5a27758c418dbaa6608567bfc6b06bbfdf TEST Deployment Plan.docx](https://github.com/user-attachments/files/18847732/14818.-.b8574c5a27758c418dbaa6608567bfc6b06bbfdf.TEST.Deployment.Plan.docx)

**Comment by Nadin-Nader** [comment]
This has been sent to the CTM DevOps Team to be deployed to TEST under **SCTASK0490357**

**Comment by SubhaShroff** [comment]
Thank you @Nadin-Nader I tested the deployment on TEST and this ticket completely ready on TEST , with the following changes.

- removing CIP fields for Private Deployment
- Excavation Manager is made non-mandatory for CIP. 

Thank you.

**Comment by KyleErich** [comment]
@SubhaShroff To answer you last question, no we don't need that CIP help text displaying on the portal under the Private Development sub type. Thank you!

**Comment by BondRuss** [comment]
@SubhaShroff 
For question 2, I confirm I am no longer seeing them. 
For question 3, understood. 
For question 4, thank you. 
For question 5, this is the way it appears in AMANDA test, for some reason in the citizen portal test the work order # is seperated from the rest of the fields: 

![Image](https://github.com/user-attachments/assets/c0b6d96c-0efd-4339-b803-500aa115a9cd)

**Comment by SubhaShroff** [comment]
Thank you Russ, for confirming.
@Nadin-Nader  I did a fix for the following and please create a DEV deployment for the commit - 528331a7bdd3b4fd5800806d946ab01971c53c97.

1. Fix the display order of FDUO with Work Order info.
2. Fix the display order of CIP# & CIP Project Manager.

[14818 - 528331a7bdd3b4fd5800806d946ab01971c53c97 DEV Deployment Plan.docx](https://github.com/user-attachments/files/18916814/14818.-.528331a7bdd3b4fd5800806d946ab01971c53c97.DEV.Deployment.Plan.docx)

Thank you.

**Comment by Nadin-Nader** [comment]
This has been sent to the DevOps cM Team to be deployed to DEV under **SCTASK0492480**

**Comment by Nadin-Nader** [comment]
This has been sent to the DevOps cM Team to be deployed to TEST under **SCTASK0493088*8

**Comment by TracyLinder** [comment]
**Feedback From ROW:**
- The team reviewed the demo and it looks good.

**Comment by SubhaShroff** [comment]
> This has been sent to the DevOps cM Team to be deployed to TEST under **SCTASK0493088**

@Nadin-Nader  This is deployed in TEST and tested following things are fixed on TEST and ready for UAT. Thank you.

1. CIP text is removed for Private Development subtype.

![Image](https://github.com/user-attachments/assets/174a4bd9-20de-4927-b107-764da5bf5cac)

3. FDUO info fields are removed for Private Development subtype. 

4. CIP id#, CIP Project Manager are displayed in order.

![Image](https://github.com/user-attachments/assets/0131335f-90f2-4632-8d6f-2c653264fd40)

5. FDUO and Work Order# are displayed together in order for CIP subtype

![Image](https://github.com/user-attachments/assets/b8d92583-da65-4344-ba07-46ce4ed74ab9)

**Comment by SubhaShroff** [comment]
### Update:
@Nadin-Nader This is ready on TEST for UAT. Thank you.

**Comment by Nadin-Nader** [comment]
@KyleErich @BondRuss this is ready for UAT on TEST

**Comment by BondRuss** [comment]
@Nadin-Nader some UAT feedback, @KyleErich will provide additional comments:
**EX:**
- **CIP**: CIP ID and CIP PM are not designated as required but you are prompted they are required when trying to proceed. I do understand the restrictions on this just documenting. 
- **CIP**: Application fees appear to be private pay only but we have customers who will be paying with FDUO, so how can we allow the customer to have the option to choose?
- **Private Development**: for some reason, step three still has some of the FDUO fields: 

![Image](https://github.com/user-attachments/assets/d46423e8-7a95-4a1c-8af4-9c85d732704d)

**RW:**
- **CIP**: There is no free-form text box to enter the CIP ID. 
- **CIP**: Why do we need the question "Is this a CIP?" if they are submitting a CIP subtype?
- **CIP**: Application fees appear to be private pay only but we have customers who will be paying with FDUO, so how can we allow the customer to have the option to choose?
- **Private Development**:  This is still showing the CIP language under the project information section but I thought we agreed to remove that on the Private Development subtype.
- **Private Development**: Why do we need the question "Is this a CIP?" and the CIP PM free form if they are submitting a Private Development subtype?
- **Private Development**: All of the FDUO fields are present but no private developer will have an FDUO for a non-CIP project.

**Comment by SubhaShroff** [comment]
Thank you for the feedback @BondRuss.

1. RW work types is not part of this ticket.
2. Apart from that, regarding EX new subtypes feedback "_CIP: Application fees appear to be private pay only but we have customers who will be paying with FDUO, so how can we allow the customer to have the option to choose?_ " -- I am not sure about  the expected criteria. Please give more details.
3. EX- Private Development: FDUO: **I will fix those two columns.** Sorry, I should have asked clearly which fields are considered as FDUO. 

CC: @vschandramouli @KyleErich  @Nadin-Nader

**Comment by TracyLinder** [comment]
### ROW Conversation Friday 3/21

- CIP subtype (functionally) can be paid either by private payment OR COS payment, but in AMANDA-EX the COS automation is triggered by the subtype, so I asked ROW to choose either private pay, or COS automation. ROW chose COS autopay. 
- The AC is being updated; we will discuss this at next Tuesday's DTS-AMANDA meeting.

**Comment by SubhaShroff** [comment]
@TracyLinder 
 ### Here is the COS auto payment logic for the existing EX-subtypes done in two scenarios:
**1. 50075	Right-Of-Way Admin (50135 Reactivated, 50040- Extension Granted)**

IF (75001-Activate Upon Payment info value is YES) COS auto pay for 

50688	Austin Water Utility (AWU)
50690	Public Works Department PWD
50691	Other City Departments
50693	Austin Energy (AE)
50694	Austin Transportation (ATD)
50696	Watershed Protection Department WPD
50698	Corridor Program Office
50710	Transportation and Public Works (TPW)

**2. Portal Submission** 

50688	Austin Water Utility (AWU)
50691	Other City Departments
50693	Austin Energy (AE)
50696	Watershed Protection Department WPD
50698	Corridor Program Office
50710	Transportation and Public Works (TPW)

**Comment by TracyLinder** [comment]
**COS Applies To:** 

50688 Austin Water Utility (AWU)
50691 Other City Departments
50693 Austin Energy (AE)
50696 Watershed Protection Department WPD
50698 Corridor Program Office
50710 Transportation and Public Works (TPW)

**Comment by SubhaShroff** [comment]
@Nadin-Nader SCTASK0502031 - submitted for DEV deployment for the fee workflow change. Thanks.

**Comment by SubhaShroff** [comment]
@Nadin-Nader The above DEV deployment is done and since I can't test this on DEV portal, I created a TEST deployment request for this - SCTASK0502792. Thank you.

**Comment by TracyLinder** [comment]
@SubhaShroff can you please let Swapna know that the portal bug is a blocker for you? I will also comment on the[ ADO ticket](https://dev.azure.com/City-of-Austin/QA%20Automation/_boards/board/t/Citizen%20Portal/Backlog%20items?System.Tags=TPW&workitem=2238) I think is causing this bug.

**Comment by SubhaShroff** [comment]
@Nadin-Nader  @TracyLinder  Informed Swapna about the bug. Thanks.

**Comment by SubhaShroff** [comment]
@Nadin-Nader The Portal 'Additional Info' display bug is fixed by Portal team. And I tested the fee flow for CIP and Private Development subtypes. Below are the tested ones:
1. Capital Improvement Project (CIP) - 2025-029372 EX - Pending Payment  (WRONG - it should be Internet Pending)
2. Private Development - 2025-029408 EX - Pending Payment

For CIP its not right. I will check the code and will update here. Thank you.

**Comment by SubhaShroff** [comment]
> [@Nadin-Nader](https://github.com/Nadin-Nader) The Portal 'Additional Info' display bug is fixed by Portal team. And I tested the fee flow for CIP and Private Development subtypes. Below are the tested ones:
> 
> 1. Capital Improvement Project (CIP) - 2025-029372 EX - Pending Payment  (WRONG - it should be Internet Pending)
> 2. Private Development - 2025-029408 EX - Pending Payment
> 
> For CIP its not right. I will check the code and will update here. Thank you.

@Nadin-Nader SCTASK0503989 - is submitted to dev deploy the CIP fee workflow correction. Thank you.

**Comment by SubhaShroff** [comment]
@Nadin-Nader As you tried to test it on DEV portal, we couldn't proceed as it was not displaying segment property and couldn't submit permits. And as you suggested I deploying it to TEST.  Created a TEST deployment pull request. Once its reviewed I would create deployment request. Thank you.
https://github.austintexas.gov/AMANDADevOps/Amandav7_TestingPOC/pull/1637

**Comment by SubhaShroff** [comment]
@Nadin-Nader SCTASK0505153 is created for TEST promotion. Thanks.

**Comment by SubhaShroff** [comment]
@Nadin-Nader  The fee workflow update is done and its deployed in TEST. I have tested few for CIP/Private Development subtypes. Ready for UAT on TEST. Tested permits:

![Image](https://github.com/user-attachments/assets/fe7b3ced-60a6-45e8-8ade-9836ce1b1352)

**Comment by BondRuss** [comment]
@SubhaShroff @Nadin-Nader 
CC: @KyleErich 
UAT in TEST is approved!

**Comment by SubhaShroff** [comment]
Thank you Russ.
@Nadin-Nader I am trying to create a PULL request for PROD promotion. Since multiple database objects involved and finding code conflicts, taking time. I will try again on monday. We might need DevOps help. But, I will try my best to resolve. 
Thank you.

**Comment by SubhaShroff** [comment]
@Nadin-Nader 
SCTASK0510244 - PROD deployment request is created. Thank you.

**Comment by SubhaShroff** [comment]
@Nadin-Nader  @TracyLinder This ticket is deployed in PROD. It should be effective after hours. I will verify again tomorrow. Thank you.

**Comment by BondRuss** [comment]
@SubhaShroff do you have any test cases in PROD?

**Comment by SubhaShroff** [comment]
@BondRuss I haven't tested on PROD as I am not supposed. I checked only if the new subtypes are showing up on PROD portal/Amanda. Thank you.

**Comment by BondRuss** [comment]
@SubhaShroff UAT in PROD approved.
